src/main/java/net/dv8tion/jda/api/interactions/commands/CommandInteraction.java

```diff
@@ -200,4 +202,49 @@ default OptionMapping getOption(@Nonnull String name)
         List<OptionMapping> options = getOptionsByName(name);
         return options.isEmpty() ? null : options.get(0);
     }
+
+    /**
+     * Gets the slash command String for this slash command.
+     * <br>This is similar to the String you see when clicking the interaction name in the client.
+     *
+     * <p>Example return for an echo command: {@code /say echo phrase: Say this}
+     *
+     * @return The command String for this slash command
+     */
+    @Nonnull
+    default String getCommandString()
+    {
+        //Get text like the text that appears when you hover over the interaction in discord
+        StringBuilder builder = new StringBuilder();
+        builder.append("/").append(getName());
+        if (getSubcommandGroup() != null)
+            builder.append(getSubcommandGroup()).append(" ");
+        if (getSubcommandName() != null)
+            builder.append(getSubcommandName()).append(" ");
+        builder.append(" ");
+        //build options (formatted appropriately)
+        for (OptionMapping o : getOptions())
+        {
+            builder.append(o.getName()).append(": ");
+            switch (o.getType())
+            {
+            case CHANNEL:
+                builder.append("#").append(o.getAsGuildChannel().getName()).append(" ");
+                break;
+            case USER:
+                builder.append("@").append(o.getAsUser().getName()).append(" ");
+                break;
+            case ROLE:
+                builder.append("@").append(o.getAsRole().getName()).append(" ");
+                break;
+            case MENTIONABLE:
+                builder.append("M:").append(o.getAsMentionable().getIdLong()).append(" ");
```

Comment:
This is wrong. The string has to MATCH what the client show. Please do proper testing on this and actually implement the correct strings.

Reply:
Did some testing, discovered client only allows User and Role for mentionable type. Added those as well as a default